feat(apollo-client): Adds type vars to subscribeToMore for variables

timhwang21 opened the pull request:
<!--
  Thanks for filing a pull request on Apollo Client!

  A few automated bots may chime in on your PR. They are here to help
  with reviewing and ensuring Apollo Client is production ready after each
  pull request merge.

    - meteor-bot will respond asking you to sign the CLA if this is your first PR.
      It may also respond with warnings, messages, or fail the build if something is off.
      Don't worry, it'll help you to fix what is broken!

    - bundlesize is a status check to keep the footprint of Apollo Client as small as possible.

    - travis-ci will run tests, checking style of code, and generally make
      sure everything is working as expected

  Please look at the following checklist to ensure that your PR
  can be accepted quickly:
-->

### Checklist:

- [ ] If this PR is a new feature, please reference an issue where a consensus about the design was reached (not necessary for small changes)
- [x] Make sure all of the significant new logic is covered by tests

Addresses #4246.

I was having some issues installing all needed dependencies, but this is just a type change and TSC doesn't show any errors for me.

timhwang21 commented:
I wasn't able to run the test locally, and it seems like the failure is just due to me not setting the tests up locally. The only thing the test does is add some dummy variables to the query and the subscription. Would really appreciate it if a maintainer could take a look!

benjamn commented:
Two notes:
* The tests are still failing after merging `master` into this branch, so you might want to take another look?
* I merged https://github.com/apollographql/apollo-client/pull/4608 which contains very similar changes to `subscribeToMore`, though I would like to merge this PR as well, to include the tests you wrote. Rebase when you have a chance, please?

timhwang21 commented:
Sounds good -- I just rebased and dropped duplicate changes (though I kept a rename of `TVariables` to `TSubscriptionVariables` as I think it's more accurate). Will try to solve the test failure next, but was hoping you can take a look as well, since the issue is definitely with my test and me not configuring it properly.

The only thing the test is testing anyways is a situation that should typecheck...
